Sources/MongoSwift/WriteConcern.swift

```diff
@@ -131,41 +103,69 @@ public class WriteConcern: Codable {
         }
     }
 
-    /// Initializes a new `WriteConcern` by copying a `mongoc_write_concern_t`.
+    /// Initializes a new `WriteConcern` with the same values as the provided `mongoc_write_concern_t`.
     /// The caller is responsible for freeing the original `mongoc_write_concern_t`.
     internal init(from writeConcern: OpaquePointer?) {
-        self._writeConcern = mongoc_write_concern_copy(writeConcern)
-    }
+        if mongoc_write_concern_journal_is_set(writeConcern) {
+            self.journal = mongoc_write_concern_get_journal(writeConcern)
+        }
+       let number = mongoc_write_concern_get_w(writeConcern)
+        switch number {
+        case MONGOC_WRITE_CONCERN_W_DEFAULT:
+            self.w = nil
+        case MONGOC_WRITE_CONCERN_W_MAJORITY:
+            self.w = .majority
+        case MONGOC_WRITE_CONCERN_W_TAG:
+            if let wTag = mongoc_write_concern_get_wtag(writeConcern) {
+                self.w = .tag(String(cString: wTag))
+            }
+        default:
+            self.w = .number(number)
+        }
 
-    private enum CodingKeys: String, CodingKey {
-        case w, j, wtimeout
+       let wtimeout = Int64(mongoc_write_concern_get_wtimeout(writeConcern))
+       if wtimeout != 0 {
+           self.wtimeoutMS = wtimeout
+       }
     }
 
-    public required convenience init(from decoder: Decoder) throws {
-        let container = try decoder.container(keyedBy: CodingKeys.self)
-        let w = try container.decodeIfPresent(W.self, forKey: .w)
-        let journal = try container.decodeIfPresent(Bool.self, forKey: .j)
-        let wtimeoutMS = try container.decodeIfPresent(Int64.self, forKey: .wtimeout)
-        try self.init(journal: journal, w: w, wtimeoutMS: wtimeoutMS)
+    private enum CodingKeys: String, CodingKey {
+        case w, journal = "j", wtimeoutMS = "wtimeout"
     }
 
     public func encode(to encoder: Encoder) throws {
         var container = encoder.container(keyedBy: CodingKeys.self)
         try container.encodeIfPresent(self.w, forKey: .w)
         if let wtimeout = self.wtimeoutMS {
             // TODO: change wtimeout to Int64 once tix SWIFT-395 gets fixed
-            try container.encode(Int32(truncatingIfNeeded: wtimeout), forKey: .wtimeout)
+            try container.encode(Int32(truncatingIfNeeded: wtimeout), forKey: .wtimeoutMS)
```

Comment:
Note that this line is now the only reason we need to have a manual `encode` function. Otherwise we could just encode the stored properties.
For that reason I think it is worth updating the TODO comment here noting that this is the reason for our manual implementation and that we can remove this `encode` method once SWIFT-395 is fixed.